src/contractions.jl

```diff
@@ -120,34 +115,18 @@ end
 
 
 function LinearAlgebra.dot(O::AbstractMPO, ψ::AbstractMPS)
-    S = promote_type(eltype(ψ), eltype(O))
-    T = typeof(ψ)
-    ϕ = T.name.wrapper(S, length(ψ))
-    for (i, (A, B)) ∈ enumerate(zip(O, ψ))
+    MPS([
```

Comment:
This will fail with `CuMPS`, please restore previous approach.

Reply:
Same as above.